manifests/server.pp

```diff
@@ -16,7 +16,7 @@
   Hash $databases                          = {},
   $ldap_ifs                                = ['/'],
   $ldaps_ifs                               = [],
-  $ldapi_ifs                               = ['/'],
+  $ldapi_ifs                               = undef,
```

Comment:
This break the test suite: the value is passed to functions that operate on arrays or string.

Maybe we can add datatypes to these 3 parameters to enforce a correct type?  I would suggest `Array[String[1]]` or `Variant[String[1],Array[String[1]]]` if we want to be full-backward compatible.

If we only accept an array, we can pass an empty one as the default value which I guess does not require much changes to achieve your goal) with the benefit that removing the value from `ldapi_ifs` will set the variable `slapd_sockets` to `""` and not stop managing it and keep it as-is.

Otherwise, the parameter needs to be made optional (`Optional[...]`) and the value need to be tested as you are doing bellow.

Reply:
Done. The other parameter's types are not set so I don't want to introduce a Variant or Optional here.

Reply:
Yes, `Array[String[1]]` the safest and more straightforward I think, but it might break some people setup.  Since `"".empty` and `[].empty` both return true, we can leave this unchanged for now.